.github/workflows/ci.yml

```diff
@@ -23,6 +25,10 @@ jobs:
       run: ctest
       working-directory: ./build
 
+    - name: Clang-format 
+      run: make check-format 
+      working-directory: ./build
```

Comment:
don't add it on the build & test pipeline because it would be a shame to not have the coverage update just because the format is not good. What you can do is add multiple jobs in the same file: 

```yaml

jobs:
  build-and-test:
    runs-on: ubuntu-latest
    # ...
  check-format:
   runs-on:ubuntu
   steps:
     - name: Check format
       run: make check-format
```